src/librustdoc/templates/page.html

```diff
@@ -7,20 +7,20 @@
     <meta name="description" content="{{page.description}}"> {#- -#}
     <meta name="keywords" content="{{page.keywords}}"> {#- -#}
     <title>{{page.title}}</title> {#- -#}
```

Comment:
Is there a way to get Askama to automatically strip the extra space rather than having to use `{#- -#}` all over the place? This is one of my annoyances with templates, and it seems like it should theoretically be solvable.

Reply:
There is not currently. Askama is a general-purpose text templating mechanism, and the "extra space" is only useless in the particular context of writing out HTML. I don't know the exact details of how reducing whitespace in a string serialization of the HTML DOM will affect that DOM, but if there is a reliable algorithm to implement that, I'd suggest you open an issue against Askama to discuss how we might support that.

(For example, one way I might imagine doing it is by stripping any whitespace after a newline, but that doesn't remove all the whitespace the `{#- -#}` currently remove because it leaves the newlines themselves. However, if you strip the newlines themselves you would invalidate content like `<img src="foo"\nalt="foo">`. So solving this optimally seems like a decently hard problem.)

Reply:
At least stripping the extra spaces after the newline seems like it'd be good enough to me. Sometimes the newlines can be helpful for debugging the HTML too.

Reply:
I don't think I will have time to write code for this, but I'm happy to discuss how it could be implemented and review the code.

Reply:
One of the Askama team members submitted a PR for something like this today: https://github.com/djc/askama/pull/598.